src/tpm2tss-genkey.c

```diff
@@ -73,13 +74,15 @@ static const struct option long_options[] = {
     {"parent",   required_argument, 0, 'P'},
     {"keysize",  required_argument, 0, 's'},
     {"verbose",  no_argument,       0, 'v'},
+    {"symmode",  required_argument, 0, 'm'},
     {0,          0,                 0,  0 }
 };
 
 static struct opt {
     char *filename;
     TPMI_ALG_PUBLIC alg;
     TPMI_ECC_CURVE curve;
+    TPMI_ALG_SYM_MODE sym_mode;
     int exponent;
```

Comment:
The help string, the longopts and this struct use different names, i.e. mode, symmode, sym_mode. Those should be unified.